src/main/scala/rx/lang/scala/observables/AsyncOnSubscribe.scala

```diff
@@ -0,0 +1,120 @@
+package rx.lang.scala.observables
+
+import rx.annotations.Experimental
+import rx.lang.scala.{Notification, Observable, Observer}
+
+/**
+  * An utility class to create observables that start acting when subscribed to and respond correctly
+  * to back pressure requests from subscribers.
+  */
+private[scala] class AsyncOnSubscribe[S,T](val generatorF: () => S, val nextF: (S, Long, Observer[Observable[_ <: T]]) => S, val onUnsubscribeF: S => Unit) {
+  import rx.lang.scala.JavaConversions._
+
+  private[scala] val asJavaAsyncOnSubscribe = new rx.observables.AsyncOnSubscribe[S,T] {
+    override def generateState(): S = generatorF()
+    override def next(state: S, requested: Long, observer: rx.Observer[rx.Observable[_ <: T]]): S =
+      nextF(state, requested, Observer(
+        onNext = t => observer.onNext(t.asJavaObservable),
+        onError = e => observer.onError(e),
+        onCompleted = () => observer.onCompleted()
+      ))
+    override def onUnsubscribe(state: S): Unit = onUnsubscribeF(state)
+  }
+
+  private def toObservable = toScalaObservable(rx.Observable.create(asJavaAsyncOnSubscribe))
+}
+
+object AsyncOnSubscribe {
+  /**
+    * Generates an {@code Observable} that synchronously calls the provided {@code next} function
```

Comment:
Volgens mij doet `{@code` niks in scaladoc, en moet je daar backticks gebruiken net als in markdown.

http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html#markup